### PR TITLE
Add a reason attribute to filePUT and fileGET

### PR DESCRIPTION
**Pull request by bpasero**
This adds a new reason property to `filePUT` and `fileGET` that could give a little bit more information why certain file types are so popular over others. Most specifically, for the `fileGET` case we will be able to filter out those calls that did not originate from a normal editor open. To give an example, when you open a TS file, we also seem to resolve a `tsconfig.json` file from the TS extension. This file is not opened as editor but still resolved and we count a `fileGET` for this call. This seems to happen only once though.

### CODE REVIEW

File: src/vs/workbench/services/textfile/common/textFileEditorModel.ts [hunk 1]
```diff
@@ -360,10 +360,16 @@ export class TextFileEditorModel extends BaseTextEditorModel implements ITextFil
 					"fileGet" : {
 						"mimeType" : { "classification": "SystemMetaData", "purpose": "FeatureInsight" },
 						"ext": { "classification": "SystemMetaData", "purpose": "FeatureInsight" },
-						"path": { "classification": "SystemMetaData", "purpose": "FeatureInsight" }
+						"path": { "classification": "SystemMetaData", "purpose": "FeatureInsight" },
+						"reason": { "classification": "SystemMetaData", "purpose": "FeatureInsight" }
```
Comment (ramya-rao-a) [inline src/vs/workbench/services/textfile/common/textFileEditorModel.ts]: The reason here is a number and so comes under measurements not properties in the eventual payload of the telemetry event. So add the `"isMeasurement": true` in the GDPR annotation here.

**Review by ramya-rao-a — APPROVED** [review]

**Comment by seanmcbreen** [comment]
I love this idea and the approach (adding a property so we can check the delta) - is there a way to indicate an open based on a folder open that resulted in past windows that resulted in files opening again?  

e.g. I closed 15 windows last week (some JSON, Some JS, some Java) and opened that folder this week again (and they were restored)?  

You may have accounted for this already but asking anyhow.

Sean

**Comment by bpasero** [comment]
@seanmcbreen no, it would all look like a normal editor open. Let's wait to see what the current changes tell us and then tweak this further if we have to.

**Comment by ramya-rao-a** [comment]
LGTM

**Pull request merged**